features/smoke_tests/handled.feature

```diff
@@ -0,0 +1,164 @@
+    And the exception "message" equals "HandledJavaSmokeScenario"
+    And the exception "type" equals "android"
+    And the payload field "events.0.exceptions.0.stacktrace" is a non-empty array
+    And the event "unhandled" is false
+    And the event "severity" equals "warning"
+    And the event "severityReason.type" equals "handledException"
+
+    # App data
+    And the event "app.buildUUID" is not null
+    And the event "app.id" equals "com.bugsnag.android.mazerunner"
+    And the event "app.releaseStage" equals "production"
+    And the event "app.type" equals "android"
+    And the event "app.version" equals "1.1.14"
+    And the event "app.versionCode" equals 34
+    And the event "app.duration" is not null
+    And the event "app.durationInForeground" is not null
+    And the event "app.inForeground" is true
+    And the payload field "events.0.metaData.app.memoryUsage" is greater than 0
+    And the event "metaData.app.name" equals "MazeRunner"
+    And the event "metaData.app.lowMemory" is not null
+
+    # Device data
+    And the payload field "events.0.device.cpuAbi" is a non-empty array
+    And the event "device.jailbroken" is false
+    And the event "device.id" is not null
+    And the payload field "events.0.device.id" is stored as the value "device_id"
+    And the event "device.locale" is not null
+    And the event "device.manufacturer" is not null
+    And the event "device.model" is not null
+    And the event "device.osName" equals "android"
+    And the event "device.osVersion" is not null
+    And the event "device.runtimeVersions" is not null
+    And the event "device.runtimeVersions.androidApiLevel" is not null
+    And the event "device.runtimeVersions.osBuild" is not null
+    And the payload field "events.0.device.totalMemory" is greater than 0
+    And the payload field "events.0.device.freeDisk" is greater than 0
+    And the payload field "events.0.device.freeMemory" is greater than 0
+    And the event "device.orientation" equals "portrait"
+    And the event "device.time" is a timestamp
+    And the event "metaData.device.locationStatus" is not null
+    And the event "metaData.device.emulator" is false
+    And the event "metaData.device.networkAccess" is not null
+    And the event "metaData.device.charging" is not null
+    And the event "metaData.device.screenDensity" is not null
+    And the event "metaData.device.dpi" is not null
+    And the event "metaData.device.screenResolution" is not null
+    And the event "metaData.device.brand" is not null
+    And the event "metaData.device.batteryLevel" is not null
+
+    # User
+    And the event "user.id" is not null
+    And the payload field "events.0.user.id" equals the stored value "device_id"
+
+    # Breadcrumbs
+    And the event has a "state" breadcrumb named "Bugsnag loaded"
+    And the event has a "manual" breadcrumb named "HandledJavaSmokeScenario"
+    And the event "breadcrumbs.2.metaData.source" equals "BreadcrumbCallback"
```

Comment:
This will probably cause flakes because automatic breadcrumbs can be captured at any time, meaning there's no guarantee this breadcrumb will always be the element at index 2. The `event has a {string} breadcrumb named` step has an approach for this of iterating through to find a matching breadcrumb. An alternative would be to set a breadcrumb callback and discard anything we don't want.